src/vector_commitment/vector_commitment.cairo

```diff
@@ -61,7 +61,7 @@ struct VectorCommitmentWitness {
 fn vector_commit(
     ref channel: Channel, unsent_commitment: felt252, config: VectorCommitmentConfig
 ) -> VectorCommitment {
-    channel.read_felt_from_prover(unsent_commitment); // commitment is being sent
+    channel.read_truncated_hash_from_prover(unsent_commitment); // commitment is being sent
```

Comment:
Could you explain a little bit why `vector_commit` cannot use the new version of `read_felt_from_prover` but needs to stick to the old one?

Reply:
That is pretty simple, we need to mirror the functionality that is implemented in stone-prover the design decisions are not clear to me, but this is what it is, it should be unified in my opinion by we may not know smth.

Reply:
I see, let's merge that not to block further other PRs